sql/src/main/java/org/opensearch/sql/sql/parser/AstBuilder.java

```diff
@@ -161,6 +164,26 @@ public UnresolvedPlan visitFromClause(FromClauseContext ctx) {
     return result;
   }
 
+  /**
+   * Ensure NESTED function is not used in HAVING clause and fallback to legacy engine.
+   * Can remove when support is added for NESTED function in HAVING clause.
+   * @param func : Function in HAVING clause
+   */
+  private void verifySupportsCondition(UnresolvedExpression func) {
+    if (func instanceof Function) {
+      if (((Function) func).getFuncName().equalsIgnoreCase(
+          BuiltinFunctionName.NESTED.name()
+      )) {
+        throw new SyntaxCheckException(
+            "Falling back to legacy engine. Nested function is not supported in the HAVING clause."
```

Comment:
Why? `HAVING` implemented the same way as `WHERE`

Reply:
Not sure I understand your question, but I implemented the fallback mechanism in `AstBuilder` to allow for a fallback specific to the nested function used in the HAVING clause. The old fallback mechanism could not determine between clauses for the filter, and we now support nested in the WHERE clause and not the HAVING so we need to differentiate between nested used in each.

Reply:
I think if you remove this check, `nested` would be automatically supported in `HAVING`, so why not? Less complexity, more functionality.

Reply:
Good point, may just work as is. Will need to verify and add tests though, if it fits in the timeline maybe we can support in this release.